src/Arcus.Security.Secrets.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -66,7 +71,7 @@ private async Task<KeyVaultClient> GetClientAsync()
         {
             if (_keyVaultClient == null)
             {
```

Comment:
I already saw this previously, we should make this thread safe by wrapping the creation in a `Lazy`.

Reply:
Lazy doesn't support async which is a pain, but feel free to open an issue for this!